### PR TITLE
Andromeda v4.4.3 (MEV Power Enable, Test State, RCU->DMB Protocol Control Message Support, Heartbeat Allocation Fix)

### PR DESCRIPTION
**Pull request by cjchanx**

### CODE REVIEW

File: Components/FlightControl/RocketSM.cpp [hunk 1]
```diff
@@ -166,6 +169,8 @@ PreLaunch::PreLaunch()
 RocketState PreLaunch::OnEnter()
 {
     // We don't do anything upon entering prelaunch
```
Comment (Matt4890) [inline Components/FlightControl/RocketSM.cpp]: No longer true

File: Components/FlightControl/RocketSM.cpp [hunk 2]
```diff
@@ -200,15 +205,13 @@ RocketState PreLaunch::HandleNonIgnitionCommands(RocketControlCommands rcAction,
     case RSC_OPEN_DRAIN:
         GPIO::Drain::Open();
         SOAR_PRINT("Drain was opened in [ %s ] state\n", StateToString(currentState));
-        PBBRxProtocolTask::SendPBBCommand(Proto::PBBCommand::Command::PBB_OPEN_MEV);
         break;
     case RSC_CLOSE_DRAIN:
         GPIO::Drain::Close();
         SOAR_PRINT("Drain was closed in [ %s ] state\n", StateToString(currentState));
-        PBBRxProtocolTask::SendPBBCommand(Proto::PBBCommand::Command::PBB_CLOSE_MEV);
```
Comment (Matt4890) [inline Components/FlightControl/RocketSM.cpp:208]: Why were the command sends removed?

File: Components/FlightControl/RocketSM.cpp [hunk 3]
```diff
@@ -506,6 +515,9 @@ RocketState Launch::OnEnter()
     //TODO: Ensure Vent & Drain CLOSED (**Should we not ensure vent & drain are closed in ignition? and or exit of ARM?)
     //TODO: Send command to OPEN the MEV
```
Comment (Matt4890) [inline Components/FlightControl/RocketSM.cpp]: Resolved

File: Components/FlightControl/RocketSM.cpp [hunk 4]
```diff
@@ -506,6 +515,9 @@ RocketState Launch::OnEnter()
     //TODO: Ensure Vent & Drain CLOSED (**Should we not ensure vent & drain are closed in ignition? and or exit of ARM?)
```
Comment (Matt4890) [inline Components/FlightControl/RocketSM.cpp]: TODO is resolved. Question should be answered.

File: Components/FlightControl/TelemetryTask.cpp [hunk 5]
```diff
@@ -111,7 +110,6 @@ void TelemetryTask::SendVentDrainStatus()
     Proto::CombustionControlStatus gpioMsg;
     gpioMsg.set_drain_open(GPIO::Drain::IsOpen());
     gpioMsg.set_vent_open(GPIO::Vent::IsOpen());
-    gpioMsg.set_main_engine_valve_open(false); // This needs to be changed to either a) record the MEV status from the PBB, or b) not have thsi field for the DMBGpioStatus
```
Comment (Matt4890) [inline Components/FlightControl/TelemetryTask.cpp:114]: Should there be a replacement call here? Or is it just meant to be removed?

File: Components/SoarDebug/DebugTask.cpp [hunk 6]
```diff
@@ -182,12 +182,24 @@ void DebugTask::HandleDebugMessage(const char* msg)
...
+    	GPIO::MEV_EN::On();
+    }
+    else if (strcmp(msg, "mev disable") == 0) {
+    	GPIO::MEV_EN::Off();
+    }
+    else if (strcmp(msg, "mev close") == 0) {
+    	PBBRxProtocolTask::SendPBBCommand(Proto::PBBCommand::Command::PBB_CLOSE_MEV);
+    }
+    else if (strcmp(msg, "mev open") == 0) {
+    	//TODO: Remember to remove / make sure not enabled in final code
```
Comment (Matt4890) [inline Components/SoarDebug/DebugTask.cpp:195]: This is scary

**Review by Matt4890 — COMMENTED** [review]
Just a few questions

File: Components/FlightControl/RocketSM.cpp (hunk 1 above)
Comment (cjchanx) [reply Components/FlightControl/RocketSM.cpp]: Removed

**Review by Matt4890 — APPROVED** [review]

File: Components/FlightControl/RocketSM.cpp (hunk 1 above)
Comment (shivamdesai04) [reply Components/FlightControl/RocketSM.cpp]: agreed

**Pull request merged**